### PR TITLE
[REF] Rename Data Sets to Datasets

### LINKED ISSUE #2762: Rename Data Sets and Geo Map
##### Orange version
3.7.dev

##### Issue
Rename Data Sets to Datasets.
Avoid duplicated names for Geo Map.

##### Reasoning
While Datasets does not appear in most contemporary dictionaries, it does appear more frequently in an academic and technical setting. It is very likely the word has not made it into dictionaries yet since the coinage is fairly new. I vote for Datasets.

Name Geo Map appears in three places (core, Geo, and Text). I vote to rename the widgets.
Geo Map from core can either be removed in favour of Geo add-on.
If Geo Map from core is retained, perhaps Geo Map from Geo add-on could be renamed to simply Map, since we are already within Geo add-on and the name is implied.
Geo Map from Text should be renamed to Map Frequency (Frequency Map?) or Text Map or Doc Map (since it maps corpus on a map).

Suggestions accepted.

### PR DESCRIPTION
**Pull request by jerneju**
##### Issue
Fixes #2762 .

##### Includes
- [X] Code changes
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by ajdapretnar** [comment]
I am fine with renaming. What have we finally agreed on, @lanzagar? I think there's a problem with refactoring this more broadly. Should we keep it constrained just to this particular widget?

**Comment by jerneju** [comment]
@ajdapretnar , @lanzagar : Now everywhere is renamed.

**Comment by lanzagar** [comment]
I started scrolling through the changes of 164 changed files and after about 37 sec I spotted a bug (hint - try adding the Data Info widget to the canvas).
I propose a challenge for anybody interested - we can even have a leaderboard - how fast can _you_ find a bug and how many are there?

Sorry for some sarcasm.
Otherwise I am fine with using this other form of spelling and will not object to someone merging this PR once all the bugs are fixed ;)

(but please don't merge before the 3.9 release)

File: Orange/evaluation/scoring.py [hunk 1]
```diff
@@ -349,7 +349,7 @@ def graph_ranks(avranks, names, cd=None, cdmethod=None, lowv=None, highv=None,
     """
     Draws a CD graph, which is used to display  the differences in methods'
     performance. See Janez Demsar, Statistical Comparisons of Classifiers over
-    Multiple Data Sets, 7(Jan):1--30, 2006.
+    Multiple Datasets, 7(Jan):1--30, 2006.
```
Comment (janezd) [inline Orange/evaluation/scoring.py]: Please inform the Journal of Machine Learning Research that we decided to change the title of the paper.

File: Orange/projection/freeviz.py [hunk 2]
```diff
@@ -269,7 +269,7 @@ def freeviz_gradient(cls, X, y, embedding, p=1, weights=None, is_class_discrete=
 
         .. [1] Janez Demsar, Gregor Leban, Blaz Zupan
                FreeViz - An Intelligent Visualization Approach for Class-Labeled
-               Multidimensional Data Sets, Proceedings of IDAMAP 2005, Edinburgh.
+               Multidimensional Datasets, Proceedings of IDAMAP 2005, Edinburgh.
```
Comment (janezd) [inline Orange/projection/freeviz.py]: Same here.

File: Orange/projection/freeviz.py [hunk 3]
```diff
@@ -347,7 +347,7 @@ def freeviz(cls, X, y, weights=None, center=True, scale=True, dim=2, p=1,
...
```
Comment (janezd) [inline Orange/projection/freeviz.py]: And here.

**Comment by janezd** [comment]
I won't strongly object to renaming, but it will require some (or a lot of) getting used to. I'm not merging it, though. :)

(I just noticed that, incidentally, #2856 changes the same number of files. :))

**Comment by ajdapretnar** [comment]
I have a feeling this PR introduces more issues than it solves. Perhaps we can stick with Data Sets, it's not _wrong_.

**Comment by jerneju** [comment]
Click:
https://trends.google.com/trends/explore?q=datasets,data%20sets

**Comment by astaric** [comment]
Separate commits made this much easier to review. Thanks

**Pull request merged**